src/middleware/company.authentication.middleware.ts

```diff
@@ -0,0 +1,32 @@
+import { Request, Response, NextFunction } from 'express';
+import { logger } from '../utils/logger';
+import { authMiddleware, AuthOptions } from "@companieshouse/web-security-node";
+import { CHS_URL, OVERSEAS_ENTITY_PRESENTER_URL } from '../config';
+import { getApplicationData } from "../utils/application.data";
+import { ApplicationData } from "../model";
+import { EntityNumberKey } from "../model/data.types.model";
+
+export const companyAuthentication = (req: Request, res: Response, next: NextFunction) => {
+  try {
+    logger.debugRequest(req, `COMPANY AUTHENTICATION REQUEST`);
+
+    const appData: ApplicationData = getApplicationData(req.session);
+    const entityNumber: string = appData?.[EntityNumberKey] as string;
+
+    if (entityNumber) {
+      const authMiddlewareConfig: AuthOptions = {
+        chsWebUrl: CHS_URL,
+        returnUrl: OVERSEAS_ENTITY_PRESENTER_URL,
```

Comment:
Maybe this should go back to the Overseas Entity Query page? To allow another OE number to be entered (assuming they somehow entered & confirmed the wrong one). At least it would show them the number again...

Reply:
@AndreKramer7654 , validation on this page should not allow this to happen , so ideally should never occur, just added security....